lib/index.js

```diff
@@ -199,6 +200,17 @@ exports.Socket = function (type) {
     }
   }
 
+  if (this.type == 'pull') {
+    this._pause = false;
```

Comment:
I still think we're better off with this property always existing.
V8 "classes" behave better when they always have the same set of properties.
I really don't mind the "useless" boolean test in non-pull-sockets. I guarantee you you will not be able to measure a difference and it keeps the codebase simpler.